### PR TITLE
🌱 GH Actions

### LINKED ISSUE #1752: Migrate to travis-ci.com
Our current CI on travis-ci.org will be EOL on December 31, 2020. 

travis-ci.com will still have free builds for open source projects but we need to migrate.

More info:

- [the newsletter](https://mailchi.mp/3d439eeb1098/travis-ciorg-is-moving-to-travis-cicom)
- [migration docs](https://docs.travis-ci.com/user/migrate/open-source-repository-migration)

### PR DESCRIPTION
**Pull request by Adirio**
# Description
Migrate current Travis tests to GH Actions

# Motivation
- #1752
- Travis jobs have been queued for periods longer than one hour

Closes #1752

### CODE REVIEW

**Comment by Adirio** [comment]
This should be ready to be merged, all Travis tests have been moved to GH Actions

/cc @DirectXMan12 @estroz @camilamacedo86
@christopherhein

File: test.sh
Comment (camilamacedo86) [inline test.sh:28]: we could also use the --repo in this scenario. However, it is fine as well.

**Review by camilamacedo86 — APPROVED** [review]
It shows great for me 👍 

/approve

File: test.sh
Comment (Adirio) [reply test.sh:28]: Didn't remember the `--repo` flag.

**Comment by Adirio** [comment]
Just needs a LGTM label, lets see if @estroz or @DirectXMan12 think it's ok.

File: .github/workflows/ci.yml
Comment (estroz) [inline .github/workflows/ci.yml:39]: It also tries to remove controller-gen iirc, is that not a problem?

**Comment by christopherhein** [comment]
Non-official but this lgtm as well 🎉  great work!

**Review by estroz — APPROVED** [review]
@Adirio great work!

One thing to be aware of with github actions: restarting one job requires all jobs be re-run. Overall the speed/reliability benefits of actions outweigh this inadequacy, and this feature should be implemented at some point since a lot of users have [requested](https://github.community/t/ability-to-rerun-just-a-single-job-in-a-workflow/17234) it, so this should be merged.

/lgtm

File: .github/workflows/ci.yml
Comment (Adirio) [reply .github/workflows/ci.yml:39]: No, the problem is that the base virtual environment comes with `kustomize` installed under `/usr/local/bin` which requires `sudo` privileges to remove. The versions installed by us are fine but the first time the test is executed `rm -f $(command -v kustomize)` finds this one and can't remove it.

**Pull request merged**
